src/node_contextify.cc

```diff
@@ -1470,19 +1489,80 @@ void ContextifyContext::ContainsModuleSyntax(
                                                    id_symbol,
                                                    try_catch);
 
-  bool found_error_message_caused_by_module_syntax = false;
+  bool should_retry_as_esm = false;
   if (try_catch.HasCaught() && !try_catch.HasTerminated()) {
     Utf8Value message_value(env->isolate(), try_catch.Message()->Get());
     auto message = message_value.ToStringView();
 
     for (const auto& error_message : esm_syntax_error_messages) {
       if (message.find(error_message) != std::string_view::npos) {
-        found_error_message_caused_by_module_syntax = true;
+        should_retry_as_esm = true;
+        break;
+      }
+    }
+
+    for (const auto& error_message : throws_only_in_cjs_error_messages) {
```

Comment:
Nit - if `should_retry_as_esm` is already true, then we don't need to do these checks.